### ISSUE TITLE
Braillex EL 80C display: EAB key presses are being repeated when NVDA is used after JAWS

### ISSUE THREAD

**Issue by irrah68** [nvaccess/nvda#16379]
### Steps to reproduce:
1. Use JAWS and close it.
2. Start NVDA and use it with your Braillex display (El 80c or EL 40c). Use the Easy Access Bar normally.
### Actual behavior:

After a while the display starts scrolling automatically, repeating your last EAB key press.
### Expected behavior:

When NVDA is started after a JAWS session, the display should work normally. An user wouldn't have to disconnect and reconnect the USB cable in order to use the display with NVDA.
#### NVDA installed/portable/running from source:
Any NVDA version which can be used with Braillex EL 80c Display. I have used installed, portable, alpha and pr versions since November 2022.
#### NVDA version:
The latest alpha, 2024rc.1 etc.
#### Windows version:
Windows 11 2023H2
#### Name and version of other software in use when reproducing the issue:
I have installed the latest driver for Braillex EL 80c from Papenmeier, 2.12.36. 
#### Does the issue still occur after restarting your computer?
Yes.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

**Comment by Adriani90** [comment]
Did you also ask Jaws developers if the Jaws process closes properly? Sometimes things may still run in background although Jaws was closed.
Can you reproduce this problem also with Narator?

**Comment by irrah68** [comment]
> Did you also ask Jaws developers if the Jaws process closes properly? Sometimes things may still run in background although Jaws was closed.

There is a process called fsatproxy.exe which seems to be running when JAWS is closed. It is running even though JAWS is not started automatically during system boot. JAWS developers have been reported about this situation by local reseller of Braillex displays and JAWS.

> Can you reproduce this problem also with Narator?

I've never used Narrator. I have read that it uses BRLTTy as a Braille driver. I used BRLTTY with NVDA for many years ago, when NVDA didn't have direct support for Caiku Albatross displays and I was also using Albatross those days.

**Comment by seanbudd** [comment]
Given the issue occurs after running JAWS we are inclined to see this as a JAWS issue. It is likely they are not correctly releasing hold on the drivers. Without being able to debug and understand what JAWS is doing, there is little we can do.
Please report this to Freedom Scientific and ask them to communicate any changes NVDA needs to make if necessary.

**Comment by burmancomp** [comment]
@irrah68 does this happen if you manually stop fsatproxy.exe from task manager?

**Comment by irrah68** [comment]
> @irrah68 does this happen if you manually stop fsatproxy.exe from task manager?

Yes.

**Comment by burmancomp** [comment]
> Given the issue occurs after running JAWS we are inclined to see this as a JAWS issue. It is likely they are not correctly releasing hold on the drivers.

Then there should/might be problems with other displays as well.

**Comment by irrah68** [comment]
Someone from Freedom Scientific wrote to Finnish reseller of Braillex displays and JAWS and a person from the firm wrote to me. Here is what Freedom Scientific said:

"The only suggestion we have at this point is to exit Jaws, power cycle the display and then start NVDA."

This means nothing, because Braillex EL 80c doesn't have a power button and I have always disconnected and reconnected the USB cable after closing JAWS and before starting NVDA.

Now the situation is that Papenmeier can reproduce this issue but they don't know how to fix it. Perhaps a solution would be a KVM adapter, if the problem can't be found somewhere in NVDA's code.

**Comment by Adriani90** [comment]
We need a confirmation that this does or does not occur with Narator. I think otherwise it is very har or impossible to find something related in NVDA code that easi.
@irrah68 I understand that you don't use narator regularly, but without testing that use case investigation will be stuck. So please provide the results with Narator.

**Comment by irrah68** [comment]
Hmm, I have read that Narrator requires BRLTTY. I used BRLTTY many years ago with NVDA, when NVDA didn't have support for Caiku Albatross displays. I purchased Braillex EL 80c later, and I don't know how BRLTTY works nowdays. The problem is, that if I install BRLTTY, I can't use JAWS with its own Braille support. It could be impossible to compare results, if I can't run JAWS, stop it, start Narrator with BRLTTY, stop Narrator and start JAWS... and so on.

But I'll try and test if possible. Let's see what will happen.

**Comment by irrah68** [comment]
Of course I meant start Narrator and close it and then start NVDA and so on. If NVDA runs without problems, the problem must be somewhere between JAWS and NVDA.

**Comment by Adriani90** [comment]
So the combination between Narator and Jaws works as expected?Von meinem iPhone gesendetAm 16.11.2024 um 13:20 schrieb irrah68 ***@***.***>:﻿
Of course I meant start Narrator and close it and then start NVDA and so on. If NVDA runs without problems, the problem must be somewhere between JAWS and NVDA.

**Comment by irrah68** [comment]
I don't know yet. I have to read some documentation, install BRLTTY and learn Narrator before I can try different combinations. This will take some time, but I'll tell how it works or doesn't work.

**Comment by irrah68** [comment]
I installed Braille display support in Narrator. I got BRLTTY to work just enough that the text "No screen" appeared on the Braille display. The navigation keys and touch cursor keys of the Braillex display did not work at all, so I could not read the content displayed by Narrator on the Braille display. All the time, the text "No screen" was displayed.

When I closed Narrator and started NVDA, the Braille keys worked normally and didn't have the same problems as with JAWS. But since I wasn't able to fully test Braillex with Narrator, I can't be absolutely sure that there won't be any problems. Most likely there aren't, and if there aren't any in the Narrator-NVDA test with these settings, it's highly likely that any issues are related to JAWS. When closing JAWS and starting NVDA, keystrokes on the Braille display will start repeating automatically. The only solution to prevent this is to disconnect the USB cable before starting NVDA or get a KVM switch adapter. But the problem will still be between JAWS and NVDA.

**Comment by Adriani90** [comment]
@irrah68 do you have any updates on this issue? Can you reproduce it with NvDA last 64bit alpha?

**Comment by irrah68** [comment]
> [@irrah68](https://github.com/irrah68) do you have any updates on this issue? Can you reproduce it with NvDA last 64bit alpha?

Yes I can. First I used JAWS, closed it and started the latest alpha version (alpha-53537,5e44446e (2026.1.0.53537)). I opened a text file and started reading. When I pressed EAB right a couple of times, the display started to scroll automatically. Nothing has changed since I opened this issue.
